### DISCUSSION TITLE
New Piece: Scan

### DISCUSSION THREAD

**Discussion by Xavman42** [DigiScore/neoscore#112] [Show and tell]
Just finished a collection of small pieces for multi-percussion! Most of the code is [here](https://github.com/Xavman42/scan---piece-for-percussion). 

To play this piece, a projector needs to be mounted 7-9 feet in the air, facing down. Video is projected onto the floor, and the percussionist places instruments in the indicated locations.

![image](https://github.com/DigiScore/neoscore/assets/79334532/5a3ca42d-77ac-4afa-bfaa-695d02c287ca)

Reticles move across the screen, and the percussionist hits an instrument when the reticle passes over it. Color indicates dynamic, and linetype indicates mallet/stick type. To assist with learning, there is a staff running across the top of the screen which may or may not be hidden in the premiere performance.

I've put together 6 little movements. Beyond using neoscore for the visuals, some AI sound generation from MusicLM was used for _nebula_, I salvaged an old saxophone and SuperCollider piece of mine for the audio in _circus_, the images in _fish_ are AI generated, the audio in _fish_ was made using VCV Rack, and the squishy sounds in _spores_ are recordings of Knox blocks! Below are some practice videos.

_acceleration_ - [https://youtu.be/XdM-PcAsUg4](https://youtu.be/XdM-PcAsUg4?fbclid=IwAR30XvaWRY2-RJoz8tHm9LHDq969ywijHSNPLyo0KQDokoN3N2bC02baMNI)
_circus_ - [https://youtu.be/ugQx6aHAuGs](https://youtu.be/ugQx6aHAuGs?fbclid=IwAR0JzLs8NAy7VpkbAFOFhxmn3HWnGe2ClnE_X_vXq1Uj6YgzL4jGv1AYnSo)
_dynamic_ - [https://youtu.be/iNozGG9jhtc](https://youtu.be/iNozGG9jhtc?fbclid=IwAR3RV-B6R1wsMsNPeAbvDbIAQUyaeV_btLUslPcfvr6oWiTItAGlxgrtzE0)
_fish_ - [https://youtu.be/8_UuQugdqI8](https://youtu.be/8_UuQugdqI8?fbclid=IwAR2kzlQXYM4dkYkUW5JqXz4k48I5THKUftWXLR9L8et2nHFd_lUNnwKiO64)
_nebula_ - [https://youtu.be/sYHYtPDFuEo](https://youtu.be/sYHYtPDFuEo?fbclid=IwAR3drpQ3TQGZm6v5D9xsmB_cVesUtMc4qRto2Z1_sh-9a-cNgQWFho8D1cs)
_spores_ - [https://youtu.be/NFffE6hejhk](https://youtu.be/NFffE6hejhk?fbclid=IwAR09gsP3qXLH__RqsxbTzjlnlcGRQRZXObTV2y9XcNLhvL1JpC7Govi0QAE)

The only technical thing I've overlooked is some video compression artifacting coming from ffmpeg - it's most obvious in _dynamic_. Now I've gotten used to it, it may even improve that particular movement.

A few movements of the piece will be premiered sometime this semester, still need to figure out the exact date. They will be premiered as part of the Turn Around concert series here in Urbana IL at the house of Patch Adams and Susan Parenti.

It is ridiculously easy to come up with new and interesting movements using this architecture - I could end up with a couple dozen movements if I'm not careful.

**Comment by ajyoon** [comment]
This is extremely cool and a very compelling compositional approach. Please share videos of the performance if/when they're available!

I would expect the compression artifacts to be fixable with a little ffmpeg tuning, but I also agree the artifacts look nice in their own way.

**Comment by Xavman42** [comment]
The piece premiered! The movements we decided to perform are acceleration, circus, dynamic, and fish.
https://youtu.be/rLl9db8vVMU?si=GfpRhKFSskDA3Y8m
